src/plasmapy/analysis/swept_langmuir/helpers.py

```diff
@@ -235,3 +238,330 @@ def sort_sweep_arrays(
+
+    return is_regular_grid
+
+
+def _merge_voltage_clusters__zero_diff_neighbors(
+    voltage: np.ndarray, current: np.ndarray
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Take the ``voltage`` and ``current`` arrays associated with a swept
+    langmuir trace and average together clusters of identical voltage
+    values.
+    """
+    # generate boolean mask for zero diff locations
+    voltage_diff = np.diff(voltage)
+    mask_zero_diff = np.isclose(voltage_diff, 0.0)
+    mask_zero_diff = np.append(mask_zero_diff, [mask_zero_diff[-1]])
+
+    # initialize new voltage and current arrays
+    new_voltage = np.full(voltage.shape, np.nan, dtype=voltage.dtype)
+    new_current = np.full(current.shape, np.nan, dtype=current.dtype)
+
+    mask = np.logical_not(mask_zero_diff)
+    new_voltage[mask] = voltage[mask]
+    new_current[mask] = current[mask]
+
+    # merge zero diff clusters
+    while np.any(mask_zero_diff):
+        volt = voltage[mask_zero_diff][0]
+        index = np.where(new_voltage == volt)[0]
+        if len(index) == 0:
+            index = np.where(mask_zero_diff)[0]
+
+        index = index[0]
+        volt_mask = np.isclose(voltage, volt)
+
+        new_voltage[index] = volt
+        new_current[index] = np.average(current[volt_mask])
+
+        mask_zero_diff[volt_mask] = False
+
+    # remove nan entries
+    mask = np.logical_not(np.isnan(new_voltage))
+    new_voltage = new_voltage[mask]
+    new_current = new_current[mask]
+
+    return new_voltage, new_current
+
+
+def _merge_voltage_clusters__within_dv(  # noqa: C901
+    voltage: np.ndarray,
+    current: np.ndarray,
+    voltage_step_size: float,
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Take the ``voltage`` and ``current`` arrays associated with a swept
+    langmuir trace and average together clusters of voltage values
+    within step size ``voltage_step_size``.
+    """
+    # initialize new voltage and current arrays
+    new_voltage = voltage.copy()
+    new_current = current.copy()
+
+    voltage_diff = np.diff(voltage)
+
+    # generate global cluster mask
+    mask1 = voltage_diff < voltage_step_size
+    mask2 = np.isclose(voltage_diff, voltage_step_size)
+    cluster_mask = np.logical_or(mask1, mask2)
+
+    # determine cluster locations
+    indices = np.where(np.diff(cluster_mask))[0]
+    if cluster_mask[0]:
+        # 1st element in voltage is in a cluster
+        indices = np.append([0], indices)
+        indices[1] = indices[1] + 1
+    else:
+        indices[0:2] = indices[0:2] + 1
+    if indices.size > 1:
+        indices[2:] = indices[2:] + 1
+    if cluster_mask[-1] and indices[-1] != voltage.size - 1:
+        # last element in voltage is in a cluster
+        indices = np.append(indices, [voltage.size - 1])
+    indices = np.reshape(indices, newshape=(int(indices.size / 2), 2))
+    # ^ using newshape kwarg for backwards compatibility, newshape kwarg
+    #   has been deprecated since numpy v2.1
+
+    # populate
```

Comment:
```suggestion
    # populate voltages and currents
```
Should we expand the comment into something like this?